### ISSUE TITLE
Castle.WcfIntegrationFacility target framework warning

### ISSUE THREAD

**Issue by tdavisjr** [castleproject/Windsor#360]
Hi,

When attempting to reference this nuget package from a web project that targets .netcore2.0, I get the following warning:

**warning NU1701: Package 'Castle.WcfIntegrationFacility 4.1.0' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.**

Can anyone explain why I'm receiving such an error since I thought this project was updated to support .NET Core recently?

I was able to get rid of the warning by updating my target framework to be .NET 4.6.2.

**Comment by fir3pho3nixx** [comment]
Yes but not the WcfIntegrationFacility. WCF to my knowledge does not run on dotnet core. Have you checked out `Wcf Connected Services`?

**Comment by tdavisjr** [comment]
If you are referring to how Visual Studio can generate SOAP client proxies, then yes. However, I don't like the code it generates so I currently rely on this facility to create them for me.

Thanks for clarifying that WCF (server side) isn't supported in .NET Core. I do believe the client side libraries are supported which is really how I wanted to use this library. Perhaps this package can be split to remove the server aspects so that the client api can be referenced as a .NET standard project.

**Comment by fir3pho3nixx** [comment]
I have never really used WCF, so I am not sure what this would entail. Do you think you could explain what features you are after specifically? It would help steer me in the right direction.

**Comment by tdavisjr** [comment]
Sure. Let's say the are WCF services that are already built and published to some url, instead of using the VS tools to add a Connected Service, I would write registration code that looks like this.

`container.Register(Component.For<IHelloService>()
.AsWcfClient(WcfEndpoint.At("http://someurl.com/HelloService.svc"))
.LifestyleTransient()`

IHelloService was defined in a common assembly that both the client and the server references. IHelloService was also marked with the [ServiceContract] attribute so WCF knows it exists.

Then to call this code, I would just have to inject IHelloService and call any method that is marked with an [OperationContract] attribute. 

So, I'm really after that .**AsWcfClient**(...) helper. I'm not trying to write services using WCF anymore (moved to Web API); but I do frequently have to call them and this helper seems to do all the work necessary to create the SOAP client proxy.

I hope this helps. I create [this repo](https://github.com/tdavisjr/windsor-aspnet-core) with some sample code; but it's targeting the full .NET framework instead of the cross-platform .NET core due to the library incompatibility.

**Comment by fir3pho3nixx** [comment]
Good stuff. Give me some time to absorb this and get back to you.

**Comment by fir3pho3nixx** [comment]
I forked: https://github.com/fir3pho3nixx/windsor-aspnet-core/tree/connected-services

After downloading and installing: https://marketplace.visualstudio.com/items?itemName=WCFCORETEAM.VisualStudioWCFConnectedService

I Generated a proxy client for: http://localhost:50214/HelloService.svc

Then added 1 line of code to register the client and I was consuming the client in `Index.cshtml.cs` using Windsor.

What is so hard about doing this?

**Comment by fir3pho3nixx** [comment]
For posterity: https://github.com/tdavisjr/windsor-aspnet-core/pull/1

**Comment by fir3pho3nixx** [comment]
https://github.com/dotnet/announcements/issues/70

**Comment by fir3pho3nixx** [comment]
https://github.com/dotnet/announcements/issues/72
